Code/training_common_utils.py

```diff
@@ -187,11 +194,11 @@ def training_loop(
             loss_training_calc, loss_validation_calc))
         print("\tEpoch Compute Time: {}".format(
             time.time() - epoch_start))
-        if((loss_training_calc < min_train_loss) | (loss_validation_calc < min_val_loss):
+        if loss_training_calc < min_train_loss or loss_validation_calc < min_val_loss:
```

Comment:
had to adjust the syntax. my python interpreter didn't like it as is.

Reply:
as a thought...
```suggestion
        if loss_validation_calc < min_val_loss:
```

Reply:
Which version are you using? it works fine with my Python 3

Reply:
python 3.6.. 
i also realized later that we really shouldn't be basing it on loss_training_calc, since that is always  going to decrease (the whole backpropagation algorithm's convergence thing...) as the network "memorizes" the training data.